os/board/rtl8721csm/src/component/common/api/wifi_interactive_mode.c

```diff
@@ -728,43 +733,68 @@ int8_t cmd_wifi_on(WiFi_InterFace_ID_t interface_id)
 {
 	int ret;
 
-	/* Kill init thread after all init tasks done */
-	ret = wifi_on(RTW_MODE_STA);
-	if (ret != RTW_SUCCESS) {
-		ndbg("\n\rrtk_wifi_start failed\n");
-		return RTK_STATUS_ERROR;
+	if (rltk_wlan_running(WLAN0_IDX)) {
+		if (wifi_set_mode(RTW_MODE_STA) < 0) {
+			ndbg("\n\rERROR: Wifi Set Mode to STA failed!\n");
+			return RTW_ERROR;

```

Comment:
RTW_ERROR value is -1 where as RTK_STATUS_ERROR at line 747 value is 1.
So, what is the expected return value for this API in error case?